uninstall.sh

```diff
@@ -0,0 +1,348 @@
+#!/bin/bash
+set -u
+shopt -s extglob
+
+# Set up temp dir
+tmpdir=/tmp/uninstall.$$
+mkdir -p "$tmpdir" || Kernel_abort "Unable to create temp dir '$tmpdir'"
+trap '
+  rm -fr "$tmpdir"
+  # Invalidate sudo timestamp before exiting
+  /usr/bin/sudo -k
+' exit
+
+### Shell equivalents to standard Ruby methods
+
+Kernel_abort() {
+  [[ $1 ]] && echo "$1" 2>/dev/null
+  exit 1
+}
+
+String_strip() {
+  local s; for s in "$@"; do
```

Comment:
`local s` might be better to be in a separate line. Readability and consistency with `Dir_children`.

Reply:
Good point, will fix.